src/liboslexec/opmatrix.cpp

```diff
@@ -198,20 +198,24 @@ osl_get_inverse_matrix(void* sg_, void* r, const char* to)
 // Implemented by the renderer
 #    define OSL_SHADEOP_EXPORT extern "C" OSL_DLL_EXPORT
 OSL_SHADEOP_EXPORT OSL_HOSTDEVICE int
-osl_get_matrix(void* sg_, void* r, const char* from);
+__optix_enabled__osl_get_matrix(void* sg_, void* r, const char* from);
 OSL_SHADEOP_EXPORT OSL_HOSTDEVICE int
-osl_get_inverse_matrix(void* sg_, void* r, const char* to);
+__optix_enabled__osl_get_inverse_matrix(void* sg_, void* r, const char* to);
 #    undef OSL_SHADEOP_EXPORT
 #endif  // __CUDACC__
 
-
+#ifdef __CUDACC__
+#define OPTIX_ENABLED_CALL(call) __optix_enabled__##call
+#else
+#define OPTIX_ENABLED_CALL(call) call
+#endif
 
 OSL_SHADEOP OSL_HOSTDEVICE int
-osl_prepend_matrix_from(void* sg_, void* r, const char* from)
+OPTIX_ENABLED_CALL(osl_prepend_matrix_from(void* sg_, void* r, const char* from))
```

Comment:
Suggest not having OPTIX mentioned directly in the macro and just have it on the function name.
Perhaps similar to what's in
https://github.com/AcademySoftwareFoundation/OpenShadingLanguage/blob/dc6ce6c82b58e5638bf2798d9650445bbdac50f9/src/liboslexec/wide/define_opname_macros.h

Such that its clearly defined its meant to be called back into from render and from OSL JIT:
`__OSL_RS_OP(prepend_matrix_from)(void* sg_, void* r, const char* from)`

where 

```
#ifdef __CUDACC__
#define __OSL_RS_OP(call) __OSL_CONCAT3(__optix_enabled__,osl_,call)
#else
#define __OSL_RS_OP(call) __OSL_CONCAT2(osl_,call)
#endif
```